lib/elixir/lib/json.ex

```diff
@@ -0,0 +1,281 @@
+    vars = Macro.generate_arguments(length(fields), __MODULE__)
+    kv = Enum.zip(fields, vars)
+
+    {io, _prefix} =
+      Enum.flat_map_reduce(kv, ?{, fn {field, value}, prefix ->
+        key = IO.iodata_to_binary([prefix, :elixir_json.encode_binary(Atom.to_string(field)), ?:])
+        {[key, quote(do: encoder.(unquote(value), encoder))], ?,}
+      end)
+
+    io = if io == [], do: "{}", else: io ++ [?}]
+
+    quote do
+      defimpl JSON.Encoder, for: unquote(module) do
+        def encode(%{unquote_splicing(kv)}, encoder) do
+          unquote(io)
+        end
+      end
+    end
+  end
+
+  defp fields_to_encode(fields, opts) do
+    cond do
+      only = Keyword.get(opts, :only) ->
+        case only -- fields do
+          [] ->
+            only
+
+          error_keys ->
+            raise ArgumentError,
+                  "unknown struct fields #{inspect(error_keys)} specified in :only. Expected one of: " <>
+                    "#{inspect(fields -- [:__struct__])}"
+        end
+
+      except = Keyword.get(opts, :except) ->
+        case except -- fields do
+          [] ->
+            fields -- [:__struct__ | except]
+
+          error_keys ->
+            raise ArgumentError,
+                  "unknown struct fields #{inspect(error_keys)} specified in :except. Expected one of: " <>
+                    "#{inspect(fields -- [:__struct__])}"
+        end
+
+      true ->
+        fields -- [:__struct__]
+    end
+  end
+
+  @doc """
+  A function invoked to encode the given term.
+  """
+  def encode(term, encoder)
+end
+
+defimpl JSON.Encoder, for: Atom do
+  def encode(value, encoder) do
+    case value do
+      nil -> "null"
+      true -> "true"
+      false -> "false"
+      _ -> encoder.(Atom.to_string(value), encoder)
+    end
+  end
+end
+
+defimpl JSON.Encoder, for: BitString do
+  def encode(value, _encoder) do
+    :elixir_json.encode_binary(value)
+  end
+end
+
+defimpl JSON.Encoder, for: List do
+  def encode(value, encoder) do
+    :elixir_json.encode_list(value, encoder)
+  end
+end
+
+defimpl JSON.Encoder, for: Integer do
+  def encode(value, _encoder) do
+    :elixir_json.encode_integer(value)
+  end
+end
+
+defimpl JSON.Encoder, for: Float do
+  def encode(value, _encoder) do
+    :elixir_json.encode_float(value)
+  end
+end
+
+defimpl JSON.Encoder, for: Map do
+  def encode(value, encoder) do
+    :elixir_json.encode_map(value, encoder)
+  end
+end
+
+defmodule JSON do
+  @moduledoc """
+  JSON encoding and decoding.
+
+  Both encoder and decoder fully conform to [RFC 8259](https://tools.ietf.org/html/rfc8259) and
+  [ECMA 404](https://ecma-international.org/publications-and-standards/standards/ecma-404/)
+  standards.
+
+  ## Encoding
+
+  Elixir built-in data structures are encoded to JSON as follows:
+
+  | **Elixir**             | **JSON** |
+  |------------------------|----------|
+  | `integer() \| float()` | Number   |
+  | `true \| false `       | Boolean  |
```

Comment:
this breaks IEx docs:

<img width="391" alt="image" src="https://github.com/user-attachments/assets/446eaee2-7d89-4d75-87af-0de736729b05">